engine/api/bootstrap/bootstrap.go

```diff
@@ -48,3 +55,176 @@ func InitiliazeDB(defaultValues sdk.DefaultValues, DBFunc func() *gorp.DbMap) er
+		if err != nil {
+			return sdk.WrapError(err, "MigrateActionDEPRECATEDGitClone> Cannot start transaction")
+		}
+
+		if err := MigrateActionDEPRECATEDGitClonePipeline(tx, store, p); err != nil {
+			log.Error("MigrateActionDEPRECATEDGitClone> %v", err)
+			tx.Rollback()
+			continue
+		}
+
+		if err := tx.Commit(); err != nil {
+			return sdk.WrapError(err, "MigrateActionDEPRECATEDGitClone> Cannot commit transaction")
+		}
+	}
+
+	return nil
+}
+
+// MigrateActionDEPRECATEDGitClonePipeline is the unitary function
+func MigrateActionDEPRECATEDGitClonePipeline(db gorp.SqlExecutor, store cache.Store, p action.PipelineUsingAction) error {
+	//Override the appname with the application in workflow node context if needed
+	if p.AppName == "" && p.WorkflowName != "" {
+		w, err := workflow.Load(db, store, p.ProjKey, p.WorkflowName, nil, workflow.LoadOptions{})
+		if err != nil {
+			return err
+		}
+		node := w.GetNodeByName(p.WorkflowNodeName)
+		if node == nil {
+			return sdk.ErrWorkflowNodeNotFound
+		}
+		if node.Context != nil && node.Context.Application != nil {
+			p.AppName = node.Context.Application.Name
+		}
+	}
+
+	pip, err := pipeline.LoadPipeline(db, p.ProjKey, p.PipName, true)
+	if err != nil {
+		return sdk.WrapError(err, "unable to load pipeline")
+	}
+
+	for _, s := range pip.Stages {
+		for _, j := range s.Jobs {
+			var migrateJob bool
+			for _, a := range j.Action.Actions {
+				if a.Name == DEPRECATEDGitClone {
+					log.Info("MigrateActionDEPRECATEDGitClone> Migrate %s/%s/%s(%d)", p.ProjKey, p.PipName, j.Action.Name, j.Action.ID)
+					migrateJob = true
+					break
+				}
+			}
+			if migrateJob {
+				if err := MigrateActionDEPRECATEDGitCloneJob(db, store, p.ProjKey, p.AppName, j); err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+var (
+	originalGitClone *sdk.Action
+	anAdminID        int64
+)
+
+// MigrateActionDEPRECATEDGitCloneJob is the unitary function
+func MigrateActionDEPRECATEDGitCloneJob(db gorp.SqlExecutor, store cache.Store, pkey, appName string, j sdk.Job) error {
+	mapReplacement := make(map[int]sdk.Action)
+
+	var err error
+	//Load the builtin gitclone action is needed
+	if originalGitClone == nil {
+		originalGitClone, err = action.LoadPublicAction(db, sdk.GitCloneAction)
+		if err != nil {
+			return err
+		}
+	}
+
+	//Load the first admin we can
+	if anAdminID == 0 {
+		users, err := user.LoadUsers(db)
+		if err != nil {
+			return err
+		}
+		for _, u := range users {
+			if u.Admin {
+				anAdminID = u.ID
+				break
+			}
+		}
+	}
+
+	//Load the project
+	proj, err := project.Load(db, store, pkey, nil, project.LoadOptions.WithKeys)
+	if err != nil {
+		return err
+	}
+	projKeys := proj.SSHKeys()
+
+	//Load the application
+	log.Debug("load application %s", appName)
+	app, err := application.LoadByName(db, store, pkey, appName, nil, application.LoadOptions.WithKeys)
+	if err != nil {
+		return err
+	}
+	appKeys := app.SSHKeys()
+
+	//Check all the steps of the job
```

Comment:
env keys ?

Reply:
on our CDS instance, no need to deal with them